float32 x
float32 y
float32 intensity
float32 distance
